prow/plugins/help/help.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package help
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "help"
+
+var (
+	helpLabel         = "help wanted"
+	helpRe            = regexp.MustCompile(`(?mi)^/help\s*$`)
+	helpRemoveRe      = regexp.MustCompile(`(?mi)^/remove-help\s*$`)
+	helpGuidelinesURL = "https://git.k8s.io/community/contributors/devel/help-wanted.md"
+	helpMsg           = `
+This request has been marked as needing help from a contributor.
+
+Please ensure the request meets the requirements listed [here](` + helpGuidelinesURL + `).
+
+If this request no longer meets these requirements, the label can be removed
+by commenting with the "/remove-help" command.
+`
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	// The Config field is omitted because this plugin is not configurable.
+	return &pluginhelp.PluginHelp{
+			Description: "The help plugin provides commands that add or remove the '" + helpLabel + "' label from issues.",
+			WhoCanUse:   "Anyone can trigger this plugin on a PR.",
+			Usage:       "/[remove-]help",
+			Examples:    []string{"/help", "/remove-help"},
+		},
+		nil
+}
+
+type githubClient interface {
+	BotName() (string, error)
+	CreateComment(owner, repo string, number int, comment string) error
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+}
+
+type commentPruner interface {
+	PruneComments(shouldPrune func(github.IssueComment) bool)
+}
+
+func handleGenericComment(pc plugins.PluginClient, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, pc.CommentPruner, &e)
+}
+
+func handle(gc githubClient, log *logrus.Entry, cp commentPruner, e *github.GenericCommentEvent) error {
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	commentAuthor := e.User.Login
+
+	// Determine if the issue has the help label
+	labels, err := gc.GetIssueLabels(org, repo, e.Number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repo, e.Number)
```

Comment:
The org, repo, and number are already included as log fields in the logger when it is passed to the plugin in the PluginClient so you don't need to include those fields in the log message.